### PR TITLE
[IMP] base: improve log for wrong xpath

### PR DESCRIPTION
**Pull request by JKE-be**
Get in log wrong xpath to help developer to fix it.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/tools/template_inheritance.py [hunk 1]
```diff
@@ -51,7 +53,13 @@ def locate_node(arch, spec):
     :return: a node in the source matching the spec
     """
     if spec.tag == 'xpath':
-        nodes = etree.ETXPath(spec.get('expr'))(arch)
+        expr = spec.get('expr')
+        try:
+            xPath = etree.ETXPath(expr)
+        except etree.XPathSyntaxError:
+            _logger.error("XPathSyntaxError while parsing xpath '%s'" % expr)
```
Comment (mart-e) [inline odoo/tools/template_inheritance.py]:
```suggestion
            _logger.error("XPathSyntaxError while parsing xpath '%s'", expr)
```

**Comment by JKE-be** [comment]
Local diff, to merge or not (imo, yes).

When you adapt a module without install at each modified xpath, it is hard to find the node that crash since you don't have the xpath in the log.

How to reproduce: 

Write a wrong xpath "//div[hasclass('test')", update/install
Current Error:

> odoo.tools.convert.ParseError: "Invalid predicate" while parsing filename.xml:2, near `<odoo>`

Now we have a new log to help developer to locate the error:

> odoo.tools.template_inheritance: XPathSyntaxError while parsing xpath '//div[hasclass('test')'
> odoo.tools.convert.ParseError: "Invalid predicate" while parsing filename.xml:2, near `<odoo>`



PS: code updated following suggestion, even if not sure of the benefit since it is not translatable

**Comment by JKE-be** [comment]
%r vs %s
repr(str) != str() ? Or it just to avoid the **\'** around **\'** %x **\'**  ?

**Comment by Yenthe666** [comment]
+1 I think it is an addition and it doesn't need translations as pretty much any dev can speak English :)

**Comment by mart-e** [comment]
> code updated following suggestion, even if not sure of the benefit since it is not translatable

Indeed not translated but always good to use the full API of logging method. Technically it does make a small difference:

```python
import logging
logger = logging.getLogger("foo")
logger.setLevel(logging.ERROR)
logger.warning("Foo %s and %s" % 42)  # <- this one will fail
logger.warning("Foo %s and %s", 42)  # <= this one will do nothing as wrong level
```
(To apply in more complexe logging scenarios)

>  repr(str) != str() ? Or it just to avoid the ' around ' %x '

Just to avoid the `'` indeed.

**Comment by JKE-be** [comment]
so, it is ok for you now ?

**Comment by mart-e** [comment]
Yes !

**Comment by JKE-be** [comment]
Can you r+ ? Since self review is no more really appreciated.

**Comment by mart-e** [comment]
Changed a bit the commit message to explain the problem before create_multi

@robodoo r+

**Pull request closed without merging**